### PR TITLE
Rename permissions attribute to current_user_permissions

### PR DESCRIPTION
**Pull request by lukaw3d**
Required by https://github.com/genialis/resolwe/pull/320

### CODE REVIEW

File: src/api/types/rest.ts [hunk 1]
```diff
@@ -473,7 +473,7 @@ export interface Collection extends CollectionBase {
     data: number[];
 }
 
-export function isCollection(object: Collection | SampleBase | Data): object is Collection {
+export function isCollection(object: CollectionBase | SampleBase | Data): object is CollectionBase {
     return object.hasOwnProperty('description') &&
         object.hasOwnProperty('settings') &&
         object.hasOwnProperty('data') &&
```
Comment (hadalin) [inline src/api/types/rest.ts]: `CollectionBase` does not have `data` so maybe add a comment here.

Comment (lukaw3d) [inline src/api/types/rest.ts]: How about we just add `data: any;` into CollectionBase type?

Comment (lukaw3d) [inline src/api/types/rest.ts]: Ops. that causes issues with Sample type

Comment (lukaw3d) [inline src/api/types/rest.ts]: Fixed and moved to a separate pullrequest: https://github.com/genialis/resolwe-js/pull/35
We will need to merge that pullrequest much quicker than this one

**Comment by mstajdohar** [comment]
We need this fast as the change has been merged in resolwe.

**Pull request merged**